src/server/rest/v1/service/ChargingStationService.ts

```diff
@@ -1331,31 +1337,51 @@ export default class ChargingStationService {
+      csv += i18nManager.translate('chargers.numberOfConnectors') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('general.siteArea') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('general.latitude') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('general.longitude') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.chargePointSN') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.model') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.chargeBoxSN') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.vendor') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.firmwareVersion') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.ocppVersion') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.ocppProtocol') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.lastSeen') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.lastReboot') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.maxPower') + Constants.CSV_SEPARATOR;
+      csv += i18nManager.translate('chargers.powerLimitUnit') + '\r\n';
     }
     // Content
     for (const chargingStation of chargingStations) {
-      csv += `${chargingStation.id}` + Constants.CSV_SEPARATOR;
-      csv += `${i18nManager.formatDateTime(chargingStation.createdOn, 'L')} ${i18nManager.formatDateTime(chargingStation.createdOn, 'LT')}` + Constants.CSV_SEPARATOR;
-      csv += `${chargingStation.connectors ? chargingStation.connectors.length : '0'}` + Constants.CSV_SEPARATOR;
-      csv += `${chargingStation?.siteArea?.name ? chargingStation.siteArea.name : ''}` + Constants.CSV_SEPARATOR;
+      csv += chargingStation.id + Constants.CSV_SEPARATOR;
+      if (chargingStation.createdOn) {
+        csv += i18nManager.formatDateTime(chargingStation.createdOn, 'L') + ' ' + i18nManager.formatDateTime(chargingStation.createdOn, 'LT') + Constants.CSV_SEPARATOR;
+      } else {
+        csv += i18nManager.translate('general.invalidDate') + ' ' + i18nManager.translate('general.invalidTime') + Constants.CSV_SEPARATOR;
+      }
+      csv += (chargingStation.connectors ? chargingStation.connectors.length : '0') + Constants.CSV_SEPARATOR;
+      csv += (chargingStation?.siteArea?.name ? chargingStation.siteArea.name : '') + Constants.CSV_SEPARATOR;
       if (chargingStation.coordinates && chargingStation.coordinates.length === 2) {
-        csv += `${chargingStation.coordinates[1]}` + Constants.CSV_SEPARATOR;
-        csv += `${chargingStation.coordinates[0]}` + Constants.CSV_SEPARATOR;
+        csv += chargingStation.coordinates[1] + Constants.CSV_SEPARATOR;
+        csv += chargingStation.coordinates[0] + Constants.CSV_SEPARATOR;
       } else {
-        csv += `''${Constants.CSV_SEPARATOR}''`;
+        csv += '' + Constants.CSV_SEPARATOR + '';
```

Comment:
Please use string literal, it's more readable.

Reply:
> It should be sufficient to do the following. right?
> csv += Constants.CSV_SEPARATOR + Constants.CSV_SEPARATOR;

Otherwise the number of columns will be off by one if GPS coordinates are empty and the import in other tools will likely fail.

Reply:
I let it as it is as said Jérôme we need 2 empty columns so the CSV is correctly build